modules/appeals_api/app/services/appeals_api/pdf_construction/stamper.rb

```diff
@@ -21,33 +21,45 @@ def date_stamp_path
           text: "API.VA.GOV #{@appeal.created_at.utc.strftime('%Y-%m-%d %H:%M%Z')}",
           x: 5,
           y: 782,
-          text_only: true
+          text_only: true,
+          size: 8
         )
       end
 
       def date_and_consumer_stamp_path
+        text = generate_text(24, @appeal.consumer_name)
         CentralMail::DatestampPdf.new(date_stamp_path).run(
-          text: "Submitted by #{@appeal.consumer_name} via api.va.gov",
-          x: 429,
+          text: text,
+          x: 335,
           y: 782,
-          text_only: true
+          text_only: true,
+          size: 8
         )
       end
 
       def veteran_stamp_path
         veteran_stamp_path = "#{Common::FileHelpers.random_file_path}.pdf"
         Prawn::Document.generate(veteran_stamp_path, margin: [0, 0]) do |pdf|
           pdf.text_box @appeal.stamp_text,
-                       at: [205, 790],
+                       at: [135, 789],
                        align: :center,
                        valign: :center,
                        overflow: :shrink_to_fit,
                        min_font_size: 8,
-                       width: 215,
-                       height: 10
+                       width: 195,
+                       height: 8
         end
         veteran_stamp_path
       end
+
+      def generate_text(name_max_length, consumer_name)
+        leading_space = '  ' * (name_max_length - consumer_name.length).abs
```

Comment:
I don't _think_ we need the .abs here. Please school me if I'm wrong lol!

Reply:
In one of our examples the name is so long that I got a negative number